.github/workflows/wheels.yml

```
@@ -26,6 +26,9 @@ on:
     branches:
       - main
       - maintenance/**
+  push:
+    branches:
+      - maintenance/**
```

```suggestion
    branches:
      - maintenance/**
    tags:
      - v*
```

Please check https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushbranchestagsbranches-ignoretags-ignore, only defining `branches` will not trigger `refs/tags`.